delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/client/ElasticSearchClient.scala

```diff
@@ -268,6 +270,34 @@ class ElasticSearchClient(client: HttpClient, endpoint: Uri, maxIndexPathLength:
     } yield ()
   }
 
+  /**
+    * Runs an delete by query with the passed ''query'' and ''index''. The query is run as a task and the task is
+    * requested until it finished
+    *
+    * @param query
+    *   the search query
+    * @param indices
+    *   the indices to use on search (if empty, searches in all the indices)
+    */
+  def deleteByQuery(query: JsonObject, index: IndexLabel): HttpResult[Unit] = {
+    val deleteEndpoint = (endpoint / index.value / deleteByQueryPath).withQuery(Uri.Query(defaultDeleteByQuery))
+    val req            = Post(deleteEndpoint, query).withHttpCredentials
+    for {
+      json   <- client.toJson(req)
+      taskId <- IO.fromEither(
+                  json.hcursor.get[String]("task").leftMap(_ => HttpClientStatusError(req, BadRequest, json.noSpaces))
+                )
+      taskReq = Get((endpoint / tasksPath / taskId).withQuery(Query(waitForCompletion -> "true"))).withHttpCredentials
+      _      <- client
+                  .toJson(taskReq)
+                  .retryingOnSomeErrors(
+                    updateByQueryStrategy.retryWhen,
```

Comment:
If someone changes the updateByQueryStrategy, is it obvious that it would affect the deletion code? Consider duplicating for this reason

Reply:
I removed it at it brought more trouble than value